javascriptv3/example_code/iam/scenarios/src/iam_basics.js

```diff
@@ -0,0 +1,410 @@
+/* Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+ABOUT THIS NODE.JS EXAMPLE: This example works with AWS SDK for JavaScript version 3 (v3),
+which is available at https://github.com/aws/aws-sdk-js-v3.
+
+Purpose:
+Shows how to use AWS SDK for JavaScript (v3) to create an AWS Identity and Access Management (IAM) user, assume a role,
+and perform AWS actions.
+1. Create a user who has no permissions.
+2. Create a role that grants permission to list Amazon S3 buckets for the account.
```

Comment:
2. Create a role that grants the user permission to list Amazon S3 buckets for the account.

Reply:
done